mtp_noms_ops/apps/security/views/check.py

```diff
@@ -60,9 +65,50 @@ def get_context_data(self, **kwargs):
             {'name': self.title},
         ]
         context_data[self.object_context_key] = detail_object
-
+        context_data['related_credits'] = self._get_related_credits(api_session, context_data[self.object_context_key])
         return context_data
 
+    @staticmethod
+    def _get_related_credits(api_session, detail_object):
+        # Get the sender credits
+        sender_response = api_session.get(
+            '/senders/{sender_profile_id}/credits/?{querystring}'.format(
+                sender_profile_id=detail_object['credit']['sender_profile'],
+                querystring=urlencode([
+                    ('exclude_credit__in', detail_object['credit']['id']),
+                    ('check__isnull', False),
+                    ('include_checks', True)
+                ])
+            )
+        )
+
+        sender_response.raise_for_status()
+        sender_credits = convert_date_fields(sender_response.json().get('results'), include_nested=True)
+
+        prisoner_response = api_session.get(
+            '/prisoners/{prisoner_profile_id}/credits/?{querystring}'.format(
+                prisoner_profile_id=detail_object['credit']['prisoner_profile'],
+                querystring=urlencode([
+                    (
+                        'exclude_credit__in',
+                        # Exclude any credits displayed as part of sender credits, to prevent duplication where
+                        # both prisoner and sender same as the credit in question
+                        ','.join([str(detail_object['credit']['id'])] + [str(c['id']) for c in sender_credits])
+                    ),
+                    ('check__isnull', False),
+                    ('include_checks', True)
+                ])
+            )
+        )
+        prisoner_response.raise_for_status()
+        prisoner_credits = convert_date_fields(prisoner_response.json().get('results'), include_nested=True)
+
+        return sorted(
+            prisoner_credits + sender_credits,
+            key=lambda c: c['started_at'],
```

Comment:
Sure,  i'm happy to use either as a notion of chronology

Reply:
What should the behaviour be for checks that were had a null`actioned_at` date because they're also still pending review, do we just want them at the top of this table?

Reply:
un-actioned ones should not appear in my opinion. cos the section is "Credit **decisions** matching debit card or prisoner"

Reply:
for the checks api endpoint, there's an `actioned_by` filter… so i figure we need an equivalent for the security credit filter